src/os/bluestore/BlueStore.h

```diff
@@ -1563,6 +1563,10 @@ class BlueStore : public ObjectStore,
 
     inline void set_state(state_t s) {
        state = s;
+
+       if (osd_op)
+         osd_op->mark_event(get_state_name());
```

Comment:
Adding this code changed set_state from perfectly inlined to separate function.
While this is not your fault, and overall impact is still acceptable, how about following:
1) add "unlikely" to condition
2) add get_state_name variant that retuerns std:string_view to. Problem with conversion from const char* to string_view is that length() has to be invoked.
These are of course optional and do not affect PR's usefulness.

Reply:
OK

Reply:
>     1. unlikely

@aclamk 
For the following code, there will be 'error: cannot convert'.

```cpp
    inline void set_state(state_t s) {
       state = s;
       if (unlikely(tracer)) {
         tracer->mark_tracepoint(get_state_name());
       }
    }
```

```text
/home/hehe/work/gits/ceph.yhg/src/os/bluestore/BlueStore.h: In member function ‘void BlueStore::TransContext::set_state(BlueStore::TransContext::state_t)’:
/home/hehe/work/gits/ceph.yhg/src/os/bluestore/BlueStore.h:1562:21: error: cannot convert ‘TrackedOpRef’ {aka ‘boost::intrusive_ptr<TrackedOp>’} to ‘long int’
 1562 |        if (unlikely(tracer)) {
/home/hehe/work/gits/ceph.yhg/src/common/likely.h:25:43: note: in definition of macro ‘unlikely’
   25 | #define unlikely(x)     __builtin_expect((x),0)
```